### PR TITLE
Added Lobby List and Lobby detail pages

### LINKED ISSUE #65: (matches) create view for allowing users to join a lobby (Lucien and Moses)
Goal: implement a lobby list page that display all join-able lobbies to any user. When clicking on a lobby title, user should be directed to the detail page of that lobby.

### LINKED ISSUE #66: (matches) create a view for lobby details
Goal: Create a view page for users to be able to view lobby details based on the view to join a lobby.

### PR DESCRIPTION
**Pull request by xylucien**
We have implemented basic html pages for the lobby list and detail pages for each lobby. They are under the URL "lobby/". For the lobby list, since I want to display it as a table element, I did not use the default ListView from django and instead wrote a customized view method for it.

### CODE REVIEW

File: src/chigame/games/views.py [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
 
 class GameListView(ListView):
     model = Game
     queryset = Game.objects.all()
     template_name = "games/game_list.html"
 
+def lobby_list(request):
+    lobbies = Lobby.objects.all()
+    context = {"object_list": lobbies}
+    return render(request, "games/lobby_list.html", context)
```
Comment (nugerhard) [inline src/chigame/games/views.py:15]: Maybe worth writing as a class for consistency, maybe not. I'm good with this though.

File: src/chigame/games/views.py [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+
+class ViewLobbyDetails(ListView):
```
Comment (nugerhard) [inline src/chigame/games/views.py]: I think this should inherit from DetailView vs. ListView. That way, if we go to a lobby/pk/ where a lobby with this pk doesn't exist, we get a 404.

**Review by nugerhard — CHANGES_REQUESTED** [review]
Code looks good, displays fine when I run it locally. There's only one change I suggest, and that's to have ViewLobbyDetails inherit from DetailView vs. ListView. See my note in the code. After that, gtg.

File: src/chigame/games/views.py (hunk 1 above)
Comment (elizabethli31) [reply src/chigame/games/views.py:15]: Either way (function or class) works.

**Review by elizabethli31 — CHANGES_REQUESTED** [review]
List view looks fine, but I'm not seeing anything for lobby_details. I believe you need to use DetailView and add a context_object_name.

**Comment by agustinmoses** [comment]
Updated files so that they reflect suggested changes. Namely, changed the class ViewLobbyDetails to a function instead in views.py, and updated the URL in urls.py so that they reflect the changes in views.py

**Review by nugerhard — CHANGES_REQUESTED** [review]
This code isn't functional. Please test before doing a PR. See guidelines on how to fix in slack channel.

**Review by nugerhard — APPROVED** [review]
Approved in error

**Review by nugerhard — CHANGES_REQUESTED** [review]
You set `context_object_name = "lobby_detail"` in the LobbyDetail class but then refer to `lobby` in the actual html template.

**Review by nugerhard — APPROVED** [review]
Works for me

**Review by elizabethli31 — APPROVED** [review]
LGTM! Good job guys. Can you update this branch with dev and commit? I'll merge once you've done that

**Comment by xylucien** [comment]
Lobby list view is now available. For each lobby, a detailed list is also implemented. In the future we might be more careful before each commit and should test around with a fixture.

**Comment by elizabethli31** [comment]
Issue Score: Excellent

Comments:
Good work on a simple detail page! For future PRs, make sure you code is 100% tested and ready to go instead of iterating multiple rounds of reviews. Also, make sure you detail exactly who did what in your closing comment.

**Pull request merged**